effects/zio/src/main/scalajvm/sttp/client4/httpclient/zio/HttpClientZioBackend.scala

```diff
@@ -52,12 +67,19 @@ class HttpClientZioBackend private (
 
   override protected def emptyBody(): ZStream[Any, Throwable, Byte] = ZStream.empty
 
-  override protected def bodyHandlerBodyToBody(p: Publisher[util.List[ByteBuffer]]): ZStream[Any, Throwable, Byte] =
+  override protected def lowLevelBodyToBody(p: Publisher[util.List[ByteBuffer]]): ZStream[Any, Throwable, Byte] =
     FlowAdapters.toPublisher(p).toZIOStream().mapConcatChunk { list =>
       val a = Chunk.fromJavaIterable(list).flatMap(_.safeRead()).toArray
       ByteArray(a, 0, a.length)
     }
 
+  override protected def cancelLowLevelBody(p: Publisher[ju.List[ByteBuffer]]): Unit = cancelPublisher(p)
+
+  override protected def ensureOnAbnormal[T](effect: Task[T])(finalizer: => Task[Unit]): Task[T] = effect.onExit {
+    exit =>
+      if (exit.isSuccess) ZIO.unit else finalizer.orDie
```

Comment:
I'm not sure if it's correct to use `.orDie` here. I don't think users will expect that sending a request might result in a defect. Defects are often considered unrecoverable and propagate until someone catches them, and in some cases can cause the entire application to crash. Maybe in this case it would be better to just ignore the error?

Reply:
There's a `.resurrect` later on, but yeah, maybe ignoring here is fine

Reply:
Oh, sorry, I somehow didn't notice the `.resurrect` call

Reply:
I change it to a log :)